### PR TITLE
Linear and Shenberg splines were added.

### PR DESCRIPTION
**Pull request by Bellator95**
This commit is devoted to the implementation of two kinds of splines: linear and Shenberg. Also, derivatives for Shenberg splines were added until the third order.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,32 +6,32 @@
...
+
+  - spline functions and theirs derivatives
+
+    ```python
+    import numpy as np
+    from numerical import splines
+    import matplotlib.pyplot as plt
+    
+    x = np.arange(0, 4., 0.05)
+    y = splines.shenberg(x)
```
[the remark this answers is not in the document]
Comment (Bellator95) [reply README.md]: OK

File: numerical/splines/definitions.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    idx = np.logical_and(x > 0, x < 1)
+    fx[idx] = -x[idx] + 1
+    idx = np.logical_or(x <= -1, x >= 1)
+    fx[idx] = 0
+    return fx
+
+
+@derivative.setup(ftype="numerical")
+def schoenberg(x: np.array) -> np.array:
+    """ One-dimensional Shenberg spline of 5-th order.
```
Comment (aartiukh) [inline numerical/splines/definitions.py]: fix `Shenberg`

File: numerical/splines/definitions.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    Spline defined on the interval (0, 3)
+    """
```
Comment (aartiukh) [inline numerical/splines/definitions.py]: Please add formulas for all cases. You can use Wiki or md file and put a ref on them here.

File: numerical/derivative.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from numerical.splines.derivatives import schoenberg_spline_derivatives
+
+
+class setup:
```
[the remark this answers is not in the document]
Comment (Bellator95) [reply numerical/derivative.py]: decorator `setup` in package `derivative`. Not really appropriate?

[the remark this answers is not in the document]
Comment (aartiukh) [reply numerical/derivative.py]: nmv, OK

File: numerical/splines/definitions.py (hunk 3 above)
Comment (Bellator95) [reply numerical/splines/definitions.py]: Good idea, but not in this commit, if you don't mind.

File: numerical/splines/definitions.py (hunk 2 above)
Comment (Bellator95) [reply numerical/splines/definitions.py]: OK

**Pull request merged**